samples/NovelRT.DotNet.Sample/Program.cs

```diff
@@ -0,0 +1,391 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root for more information.
+
+using System;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Text;
+using NovelRT.Interop;
+using static System.Net.Mime.MediaTypeNames;
+using static NovelRT.Interop.NovelRT;
+using static NovelRT.Interop.NrtBool;
+using static NovelRT.Interop.NrtResult;
+
+namespace NovelRT.DotNet.Sample
+{
+    public static unsafe class Program
+    {
+        private static NrtResult res = NRT_SUCCESS;
+        private static NrtBool booleanResult = NRT_TRUE;
+        private static NrtBool inkServiceProvided = NRT_FALSE;
+
+        private static int hMove = 1; // 1 == move right, 0 == move left
+        private static int vMove = 1; // 1 == move up, 0 == move down
+
+        // Services
+        private static IntPtr audio = IntPtr.Zero;
+        private static IntPtr input = IntPtr.Zero;
+        private static IntPtr console = IntPtr.Zero;
+        private static IntPtr dotnet = IntPtr.Zero;
+        private static IntPtr ink = IntPtr.Zero;
+        private static IntPtr timer = IntPtr.Zero;
+        private static IntPtr renderer = IntPtr.Zero;
+        private static IntPtr updateEvent = IntPtr.Zero;
+        private static IntPtr colourChange = IntPtr.Zero;
+
+        // Objects
+        private static IntPtr nChanRect = IntPtr.Zero;
+        private static IntPtr interactRect = IntPtr.Zero;
+        private static IntPtr story = IntPtr.Zero;
+
+        private static Random rand = null;
+
+        public static int Main(string[] args)
+        {
+            res = NRT_SUCCESS;
+            rand = new Random();
+
+            // Creating NovelRunner
+            IntPtr runner = Nrt_NovelRunner_create(0);
+
+            // Starting LoggingService
+            fixed (byte* customTitle = Encoding.UTF8.GetBytes("Interop"))
+            {
+                console = Nrt_LoggingService_createCustomTitle((sbyte*)customTitle);
+            }
+
+            // Getting a constant to the Resources folder
+            sbyte* execPath = Nrt_getExecutableDirPath();
+            sbyte* error = Nrt_getLastError();
+
+            if (error != null)
+            {
+                Nrt_LoggingService_logErrorLine(console, error);
+                return -1;
+            }
+
+            sbyte* path;
+
+            fixed (byte* pathPart = Encoding.UTF8.GetBytes("Resources"))
```

Comment:
The C# bindings don't have any of the "nice" surface area or helpers that take `string` yet, so need to explicitly pin and convert in the sample.

This will be improved in a later PR.